### PR TITLE
CSHARP-2836: Handle absence of 'ns' field in index specifications returned from listIndexes.

### PR DESCRIPTION
**Pull request by DmitryLukyanov**
EG: https://evergreen.mongodb.com/version/5e506c297742ae65713bbbe3

The ticket doesn't require changes from the driver side except for adding the small test

### CODE REVIEW

File: tests/MongoDB.Driver.Tests/MongoIndexManagerTests.cs [hunk 1]
```diff
@@ -0,0 +1,86 @@
...
+using MongoDB.Driver.Core.Operations;
+using Xunit;
+
+namespace MongoDB.Driver.Tests
+{
+    public class MongoIndexManagerTests
+    {
+        [Theory]
+        [ParameterAttributeData]
+        public void List_should_work_as_expected(
```
Comment (rstam) [inline tests/MongoDB.Driver.Tests/MongoIndexManagerTests.cs]: This name is not incorrect, but more commonly we would name this method:

    List_should_return_expected_result

File: tests/MongoDB.Driver.Tests/MongoIndexManagerTests.cs [hunk 2]
```diff
@@ -0,0 +1,86 @@
...
+            [Values("{ singleIndex : 1 }", "{ compoundIndex1 : 1, compoundIndex2 : 1 }")] string key,
+            [Values(false, true)] bool uniqueArgument,
```
Comment (rstam) [inline tests/MongoDB.Driver.Tests/MongoIndexManagerTests.cs]: Rename to `unique` (see comment below that avoids name conflict).

File: tests/MongoDB.Driver.Tests/MongoIndexManagerTests.cs [hunk 3]
```diff
@@ -0,0 +1,86 @@
...
+                    ? subject.ListAsync().GetAwaiter().GetResult()
+                    : subject.List();
+            var indexes = indexesCursor.ToList();
+
+            indexes.Count.Should().Be(2);
+            AssertIndex(collection.CollectionNamespace, indexes[0], "_id_");
+            var indexName = IndexNameHelper.GetIndexName(indexKeyDocument);
+            AssertIndex(collection.CollectionNamespace, indexes[1], indexName, unique : uniqueArgument);
+
+            void AssertIndex(CollectionNamespace collectionNamespace, BsonDocument index, string nameValue, bool unique = false)
```
Comment (rstam) [inline tests/MongoDB.Driver.Tests/MongoIndexManagerTests.cs]: Rename `nameValue` to `expectedName`.

Rename `unique` to `exepctedUnique`.

File: tests/MongoDB.Driver.Tests/MongoIndexManagerTests.cs [hunk 4]
```diff
@@ -0,0 +1,86 @@
...
+            {
+                index["name"].Should().Be(BsonValue.Create(nameValue));
```
Comment (rstam) [inline tests/MongoDB.Driver.Tests/MongoIndexManagerTests.cs]: This line is not incorrect but more commonly we would have written this as:

    index["name"].AsString.Should().Be(expectedIndexName);

File: tests/MongoDB.Driver.Tests/MongoIndexManagerTests.cs [hunk 5]
```diff
@@ -0,0 +1,86 @@
...
+                    index["unique"].AsBoolean.Should().BeTrue();
+                }
+                else
+                {
+                    index.Contains("unique").Should().BeFalse();
+                }
+
+                if (CoreTestConfiguration.ServerVersion < new SemanticVersion(4, 3, 0))
+                {
+                    index["ns"].Should().Be(BsonValue.Create(collectionNamespace.ToString()));
```
Comment (rstam) [inline tests/MongoDB.Driver.Tests/MongoIndexManagerTests.cs]: More commonly written as:

    index["ns"].AsString.Should().Be(collectionNamespace.ToString());

**Review by rstam — CHANGES_REQUESTED** [review]
Minor renamings requested.

File: tests/MongoDB.Driver.Tests/MongoIndexManagerTests.cs (hunk 1 above)
Comment (DmitryLukyanov) [reply tests/MongoDB.Driver.Tests/MongoIndexManagerTests.cs]: done

File: tests/MongoDB.Driver.Tests/MongoIndexManagerTests.cs (hunk 2 above)
Comment (DmitryLukyanov) [reply tests/MongoDB.Driver.Tests/MongoIndexManagerTests.cs]: done

File: tests/MongoDB.Driver.Tests/MongoIndexManagerTests.cs (hunk 3 above)
Comment (DmitryLukyanov) [reply tests/MongoDB.Driver.Tests/MongoIndexManagerTests.cs]: done

File: tests/MongoDB.Driver.Tests/MongoIndexManagerTests.cs (hunk 4 above)
Comment (DmitryLukyanov) [reply tests/MongoDB.Driver.Tests/MongoIndexManagerTests.cs]: done

File: tests/MongoDB.Driver.Tests/MongoIndexManagerTests.cs (hunk 5 above)
Comment (DmitryLukyanov) [reply tests/MongoDB.Driver.Tests/MongoIndexManagerTests.cs]: done

**Review by DmitryLukyanov — COMMENTED** [review]
ready for review.

**Review by rstam — APPROVED** [review]
LGTM

**Pull request closed without merging**